internal/common/baseclient/iigo.go

```diff
@@ -52,8 +52,26 @@ func (c *BaseClient) TaxTaken(shared.Resources) {
 
 // GetTaxContribution gives value of how much the island wants to pay in taxes
 func (c *BaseClient) GetTaxContribution() shared.Resources {
-	// TODO: Implement common pool contribution greater than or equal to tax.
-	return 0
+	// If no new communication was received and there is no rule governing taxation, it will still recieve last valid tax
+	gameState := c.ServerReadHandle.GetGameState()
```

Comment:
Maybe rename to clientGameState, so it avoids confusion with the gameState in IIGO?